Tasks/JenkinsQueueJob/jenkinsqueuejobtask.ts

```diff
@@ -9,196 +9,87 @@ import tl = require('vsts-task-lib/task');
-        return 'Success';
-    } else if (resultCode == 'UNSTABLE') {
-        return 'Unstable';
-    } else if (resultCode == 'FAILURE') {
-        return 'Failure';
-    } else if (resultCode == 'NOT_BUILT') {
-        return 'Not built';
-    } else if (resultCode == 'ABORTED') {
-        return 'Aborted';
-    } else {
-        return resultCode;
+    jobQueueUrl: string = util.addUrlSegment(this.serverEndpointUrl, '/job/' + this.jobName) + ((this.parameterizedJob) ? '/buildWithParameters?delay=0sec' : '/build?delay=0sec');
+
+    constructor() {
+        tl.debug('serverEndpointUrl=' + this.serverEndpointUrl);
+        tl.debug('jobQueueUrl=' + this.jobQueueUrl);
     }
 }
 
-function checkSuccess() {
-    var resultUrl = jenkinsExecutableUrl + 'api/json';
-    tl.debug('Tracking completion status of job: ' + resultUrl);
-    request.get({ url: resultUrl }, function callBack(err, httpResponse, body) {
-        if (err) {
-            tl.setResult(tl.TaskResult.Failed, err);
-        } else if (httpResponse.statusCode != 200) {
-            failReturnCode(httpResponse, 'Job progress tracking failed to read job result');
-        } else {
-            var parsedBody = JSON.parse(body);
-            var resultCode = parsedBody.result;
-            if (resultCode) {
-                resultCode = resultCode.toUpperCase(); 
-                var resultStr = getResultString(resultCode);
-                tl.debug(resultUrl + ' resultCode: ' + resultCode + ' resultStr: ' + resultStr);
-            
-                var completionMessage = 'Jenkins job: ' + resultCode + ' ' + jobName + ' ' + jenkinsExecutableUrl;
-                if (resultCode == "SUCCESS" || resultCode == 'UNSTABLE') {
-                    createLinkAndFinish(tl.TaskResult.Succeeded, resultStr, completionMessage);
-                } else {
-                    createLinkAndFinish(tl.TaskResult.Failed, resultStr, completionMessage);
-                }
-            } else {
-                // result not updated yet -- keep trying
-                setTimeout(function () {
-                    checkSuccess();
-                }, captureConsolePollInterval);
-            }
+/**
+ * Supported parameter types: boolean, string, choice, password
+ * 
+ * - If a parameter is not defined by Jenkins it is fine to pass it anyway
+ * - Anything passed to a boolean parameter other than 'true' (case insenstive) becomes false.
+ * - Invalid choice parameters result in a 500 response.
+ * 
+ */
+function parseJobParameters(jobParameters: string[]) {
+    var formData = {};
+    for (var i = 0; i < jobParameters.length; i++) {
+        var paramLine = jobParameters[i];
+        var splitIndex = paramLine.indexOf('=');
+        if (splitIndex <= 0) { // either no paramValue (-1), or no paramName (0)
+            throw 'Job parameters should be specified as "parameterName=parameterValue" with one name, value pair per line. Invalid parameter line: ' + paramLine;
         }
```

Comment:
fixed the parsing to add a .trim().  This entire method is with a catch-all catch block which handles all exceptions (and fails the task)